framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/ClusteredWorkbenchKieServerPersistentScenarioImpl.java

```diff
@@ -168,4 +174,52 @@ public SsoDeployment getSsoDeployment() {
     public GitProvider getGitProvider() {
         return gitProvider;
     }
+
+    @Override
+    protected String overridesVersionTag() {
+        if (request.getUpgradeSettings() != null) {
+            return OpenShiftOperatorConstants.getKieOperatorUpgradeFromImageTag();
+        }
+
+        return null;
+    }
+
+    private void configureOperatorForUpgrades() {
+        if (request.getUpgradeSettings() != null) {
+            Upgrades upgrades = new Upgrades();
+            upgrades.setEnabled(true);
+            upgrades.setMinor(request.getUpgradeSettings().isMinor());
+            kieApp.getSpec().setUpgrades(upgrades);
+        }
+    }
+
+    private void upgradeDeploymentViaOperator() {
+        if (request.getUpgradeSettings() != null) {
+            logger.info("Upgrading deployment...");
+
+            int expectedKieServerVersion = kieServerDeployment.getVersion() + 1;
+            int expectedWorkbenchVersion = workbenchDeployment.getVersion() + 1;
+
+            upgradeOperatorToLatestVersion();
+
+            kieServerDeployment.waitForVersion(expectedKieServerVersion);
+            workbenchDeployment.waitForVersion(expectedWorkbenchVersion);
+
+            logger.info("Deployment upgraded.");
```

Comment:
Is there a way to make sure that new deployment really points to new images?

Reply:
Done